scripts/bin/update.dart

```diff
@@ -0,0 +1,38 @@
+#! /usr/bin/env dart
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:http/http.dart' as http;
+import 'package:scripts/src/dockerfile.dart';
+import 'package:scripts/src/versions.dart';
+
+/// Downloads the latest versions on the Dart stable and beta channels and the
+/// relevant SHA256 checksums. For all versions that haven't changed, the script
+/// verifies that the checksums are still matching.
+void main() async {
+  await update(const LocalFileSystem(), http.read);
```

Comment:
You might want to `import 'package:http/http_retry.dart'` and do:

```suggestion
  final c = RetryClient(http.Client());
  try {
    await update(const LocalFileSystem(), c.read);
  } finally {
    await c.close();
  }
```

That way you get retries and connection reuse (that might not matter since you only do one request or so), but retries is a very good idea when talking to GCS.

Reply:
Thanks, used the RetryClient everywhere, it still creates a new client for every request but that's probably not worth fixing for the few requests this makes.